app/services/timeslot_service.rb

```diff
@@ -0,0 +1,15 @@
+class TimeslotService
+  def initialize(user)
+    @user = user
+  end
+
+  def timeslot_check_role
+    if @user&.manager?
```

Comment:
You should start with user with highest role - 'admin' then check 'manager' and for others use `else`